pallets/governance-origins/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+	use frame_support::{pallet_prelude::*, traits::Currency};
+
+	type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Currency type to limit spends per spend origin
+		type Currency: Currency<Self::AccountId>;
+		/// Maximum amount able to be spent by SmallSpender origin from treasury at once
+		#[pallet::constant]
+		type MaxSmallSpenderSpend: Get<BalanceOf<Self>>;
+		/// Maximum amount able to be spent by MediumSpender origin from treasury at once
+		#[pallet::constant]
+		type MaxMediumSpenderSpend: Get<BalanceOf<Self>>;
+		/// Maximum amount able to be spent by BigSpender origin from treasury at once
+		#[pallet::constant]
+		type MaxBigSpenderSpend: Get<BalanceOf<Self>>;
+		/// Maximum amount able to be spent by Treasurer origin from treasury at once
+		#[pallet::constant]
+		type MaxTreasurerSpend: Get<BalanceOf<Self>>;
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[derive(PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo, RuntimeDebug)]
+	#[pallet::origin]
+	pub enum Origin {
+		/// Origin able to spend up to MaxTreasurerSpend from the treasury at once.
+		Treasurer,
+		/// Origin able to cancel referenda.
+		ReferendumCanceller,
+		/// Origin able to kill referenda.
+		ReferendumKiller,
+		/// Origin able to spend up to MaxSmallSpenderSpend from the treasury at once.
+		SmallSpender,
+		/// Origin able to spend up to MaxMediumSpenderSpend from the treasury at once.
+		MediumSpender,
+		/// Origin able to spend up to MaxBigSpenderSpend from the treasury at once.
+		BigSpender,
+		/// Origin able to dispatch a whitelisted call.
+		WhitelistedCaller,
+	}
+
+	macro_rules! decl_unit_ensures {
+		( $name:ident: $success_type:ty = $success:expr ) => {
+			pub struct $name;
+			impl<O: Into<Result<Origin, O>> + From<Origin>>
+				EnsureOrigin<O> for $name
+			{
+				type Success = $success_type;
+				fn try_origin(o: O) -> Result<Self::Success, O> {
+					o.into().and_then(|o| match o {
+						Origin::$name => Ok($success),
+						r => Err(O::from(r)),
+					})
+				}
+				#[cfg(feature = "runtime-benchmarks")]
+				fn try_successful_origin() -> Result<O, ()> {
+					Ok(O::from(Origin::$name))
+				}
+			}
+		};
+		( $name:ident ) => { decl_unit_ensures! { $name : () = () } };
+		( $name:ident: $success_type:ty = $success:expr, $( $rest:tt )* ) => {
+			decl_unit_ensures! { $name: $success_type = $success }
+			decl_unit_ensures! { $( $rest )* }
+		};
+		( $name:ident, $( $rest:tt )* ) => {
+			decl_unit_ensures! { $name }
+			decl_unit_ensures! { $( $rest )* }
+		};
+		() => {}
+	}
+	decl_unit_ensures!(ReferendumCanceller, ReferendumKiller, WhitelistedCaller,);
+
+	macro_rules! decl_ensure {
+		(
+			$vis:vis type $name:ident: EnsureOrigin<Success = $success_type:ty> {
+				$( $item:ident = $success:expr, )*
+			}
+		) => {
+			$vis struct $name<T>(PhantomData<T>);
```

Comment:
```suggestion
			$vis struct $name<T>(sp_core::PhantomData<T>);
```
maybe the reason for std leaking

Reply:
nope